### ISSUE TITLE
Refine etcdctl snapshot restore error logs

### ISSUE THREAD

**Issue by hongchaodeng** [coreos/etcd#9290]
Continue discussion: https://github.com/coreos/etcd/issues/9179

The following lines are always printed to stderr
```
2018-01-19 23:42:54.398661 I | pkg/netutil: resolving test-etcd-backup-restore-0-0000.test-etcd-backup-restore-0.e2e-etcd-operator-flaketest-580.svc:2380 to 10.28.5.36:2380
2018-01-19 23:42:54.402454 I | pkg/netutil: resolving test-etcd-backup-restore-0-0000.test-etcd-backup-restore-0.e2e-etcd-operator-flaketest-580.svc:2380 to 10.28.4.183:2380
```

It has two issues:
- It causes confusion. These seems normal message. As a user I wouldn't expect to see it on stderr for successful run.
- For DNS resolving issues, the DNS and IP were already included in the error message: https://github.com/coreos/etcd/pull/9210. So the messages are duplicate.

**Comment by xiang90** [comment]
> It causes confusion. These seems normal message. As a user I wouldn't expect to see it on stderr for successful run.

No. This is the whole point of why stderr exists. Users should only redirect stdout if they only care about the final result rather than any informational or diagnostic messages.

**Comment by hongchaodeng** [comment]
@xiang90 I understand your argument. But let's be honest that this mixing of diagnostic information and user-facing error message is confusing.

If etcdctl runs successfully, it logs what it did like writing files; If it fails, it logs why it failed. In this case, `Error:  --initial-cluster ...`. Isn't it better to separate diagnostic and user messages?

As a side note, stdout/stderr doesn't really matter.
